gnovm/pkg/gnolang/machine.go

```diff
@@ -2002,8 +2005,9 @@ func (m *Machine) Panic(ex TypedValue) {
 	m.Exceptions = append(
 		m.Exceptions,
 		Exception{
-			Value: ex,
-			Frame: m.MustLastCallFrame(1),
+			Value:      ex,
+			Frame:      m.MustLastCallFrame(1),
+			Stacktrace: m.Stacktrace(),
```

Comment:
I don't think the trace should be produced here.  The reason I do not think this is the right place is because this panic might never be shown to the user or cause the running gno program to halt execution, meaning that a potentially significant amount of work to build the trace might be unnecessary.

If you look at the lines of code below this, it is doing a few things:
1. `m.PopUntilLastCallFrame()` -- this does what you'd expect - pop the frames until the machine is back to most recent function call
2. `m.PushOp(OpPanic2)` -- this op will cause the gno program to cease execution and bubble up the panic to each call frame, executing defer logic along the way and potentially recover; if it was recovered, it returns to the previous block and continues execution, not panicking and rendering the trace unused
3. `m.PushOp(OpReturnCallDefers)` -- just as in Go, when a panic occurs, any deferred functions are called before returning; this is an opportunity for the panic to be recovered

So it seems like perhaps the only thing that we may want to record in this method when appending the exception is the line number; we should still have everything else we need by the time the machine calls into `doOpPanic2`.  This way no time is wasted building a trace that may not be used.